super-ddf.c

```diff
@@ -5862,8 +5870,7 @@ static struct mdinfo *ddf_activate_spare(struct active_array *a,
 			} else if (be16_and(ddf->phys->entries[dl->pdnum].type,
 					    cpu_to_be16(DDF_Global_Spare))) {
 				is_global = 1;
-			} else if (!be16_and(ddf->phys
-					     ->entries[dl->pdnum].state,
+			} else if (!be16_and(ddf->phys->entries[dl->pdnum].state,
```

Comment:
The one side effect I can see from marking the disk Failed and Missing, back up on line 5306 in "phys_update", is that here, it seems as if it'd take it out of possibly being used as a spare here?  But, I think we can't actually hit this, because when we remove it, we're taking it all the way out.  Maybe just sanity check this.

Reply:
I like it. I like it a lot.